### PR TITLE
Initial Draft of new Framework Test Targets

### PR DESCRIPTION
**Pull request by sgoldberg-sfdc**
Here is a draft of the new framework test targets.

When running unit tests, I ran image list to validate that the unit tests were loading the dylib in the framework target.

(lldb) image list Cordova
[  0] 8AD0E773-9E10-3FAC-B0F0-F8159CE2B3FF 0x000000010e6c1000 /Users/agoldberg/Library/Developer/Xcode/DerivedData/cordova-ios-dgcbizzhftqmetgvdmvcrqodkvka/Build/Products/Debug-iphonesimulator/CordovaFrameworkApp.app/Frameworks/Cordova.framework/Cordova 

otool also indicates that the framework test bundle loads it:

agoldberg-wsm1:~ agoldberg$ otool -l CordovaFrameworkTests | grep Cordova
CordovaFrameworkTests:
         name @rpath/Cordova.framework/Cordova (offset 24)

### CODE REVIEW

**Comment by sgoldberg-sfdc** [comment]
@khawkins  @bhariharan

File: tests/CordovaLibTests/CordovaLibTests.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -65,6 +110,9 @@
 		686357B9141002F200DF4CF2 /* CDVPluginResultJSONSerializationTests.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = CDVPluginResultJSONSerializationTests.m; sourceTree = "<group>"; };
 		7E91405F17711D88002C6A3F /* CDVWebViewDelegateTests.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = CDVWebViewDelegateTests.m; sourceTree = "<group>"; };
 		7EF33BD61911ABA20048544E /* Default-568h@2x.png */ = {isa = PBXFileReference; lastKnownFileType = image.png; path = "Default-568h@2x.png"; sourceTree = "<group>"; };
+		C0FA7CAA1E4BB6420077B045 /* CordovaFrameworkApp.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = CordovaFrameworkApp.app; sourceTree = BUILT_PRODUCTS_DIR; };
+		C0FA7CAC1E4BB6B30077B045 /* Cordova.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Cordova.framework; path = "../../../../../Library/Developer/Xcode/DerivedData/cordova-ios-dgcbizzhftqmetgvdmvcrqodkvka/Build/Products/Debug-iphonesimulator/Cordova.framework"; sourceTree = "<group>"; };
```
Comment (bhariharan) [inline tests/CordovaLibTests/CordovaLibTests.xcodeproj/project.pbxproj]: Is this `DerivedData` path correct? Shouldn't it be a relative path as opposed to an absolute path?

Comment (sgoldberg-sfdc) [reply tests/CordovaLibTests/CordovaLibTests.xcodeproj/project.pbxproj]: Hmm Xcode came up with that. If I purge derived data it recreates it with the new path.  Let me see if I did something wrong there.

**Comment by sgoldberg-sfdc** [comment]
FYI: I recycled the info.plist files from the other two test targets to avoid adding new files.

**Review by bhariharan — APPROVED** [review]

**Comment by khawkins** [comment]
Like the approach!  I have one minor comment.  Because `ONLY_ACTIVE_ARCH` is set to `NO` on Debug builds for the framework, but `YES` for the static library, I get a linker warning during the framework target build, related to ignoring an i386 slice that CordovaLib builds.

Should we just remove the target-specific settings for `ONLY_ACTIVE_ARCH`, and let it inherit the YES/NO setting from the project level?

**Comment by sgoldberg-sfdc** [comment]
Good comments @khawkins . I'll experiment with the settings. Which target and platform are you building? 

There are some nuances around expectations of framework slices and I suspect whatever we will agree to is going to get different opinions from Apache and the larger community.

**Comment by khawkins** [comment]
Just building the framework target against e.g. the iPhone 6 simulator.  Only needs x86_64, gets i386, yadda yadda.

The default settings of a framework target default to YES/NO as well, so we're pretty safe there.  And once it comes to building binaries for any sort of distribution, that's where all of the interesting questions about architectures are going to matter anyway.

**Comment by sgoldberg-sfdc** [comment]
@khawkins arch setting changed

**Review by khawkins — APPROVED** [review]
Sweet, looks good to me otherwise.

**Pull request merged**